src/main/scala/org/broadinstitute/dsde/agora/server/ga4gh/Models.scala

```diff
@@ -1,10 +1,109 @@
 package org.broadinstitute.dsde.agora.server.ga4gh
 
+import org.broadinstitute.dsde.agora.server.model.{AgoraEntity, AgoraEntityType, MethodDefinition}
 import spray.json.DefaultJsonProtocol._
 import spray.json.{DeserializationException, JsString, JsValue, RootJsonFormat}
 
 object Models {
 
+  final val ID_DELIMITER = ":"
+
+  case class ToolId(namespace: String, name: String) {
+    override def toString: String = s"$namespace$ID_DELIMITER$name"
+  }
+  object ToolId {
+    def apply(entity:AgoraEntity) = {
+      assert(entity.namespace.nonEmpty, "cannot create a ToolId if entity namespace is empty")
+      assert(entity.name.nonEmpty, "cannot create a ToolId if entity name is empty")
+      new ToolId(entity.namespace.get, entity.name.get)
+    }
+    def apply(method:MethodDefinition) = {
+      assert(method.namespace.nonEmpty, "cannot create a ToolId if method definition namespace is empty")
+      assert(method.name.nonEmpty, "cannot create a ToolId if method definition name is empty")
+      new ToolId(method.namespace.get, method.name.get)
+    }
+  }
+  case class ToolClass(id: String, name: String, description: String)
+  object ToolClass {
+    def apply(entity:AgoraEntity) = fromEntityType(entity.entityType)
+    def apply(method:MethodDefinition) = fromEntityType(method.entityType)
+    def fromEntityType(entityType: Option[AgoraEntityType.EntityType]) = {
+      assert(entityType.nonEmpty, "cannot create a ToolClass if entity type is empty")
+      val str = entityType.get.toString
+      new ToolClass(str, str, "")
+    }
+  }
+
+  case class Tool(
```

Comment:
Some of these fields are not defined in the GA4GH model, but I'm guessing that's OK? My guess is that as long as we support the minimum, anything extra is gravy. This is also where we'll need alignment with the swagger spec.

Reply:
which ones are not defined in the GA4GH model? I tried to be exact: https://github.com/ga4gh/tool-registry-schemas/blob/develop/src/main/resources/swagger/ga4gh-tool-discovery.yaml#L335